tests/test_io.py

```diff
@@ -39,6 +41,43 @@ def test_load_scan_spm(load_scan: LoadScan) -> None:
     assert px_to_nm_scaling == 0.4940029296875
 
 
+def test_load_scan_asd(load_scan_asd: LoadScans) -> None:
+    """Test loading of high-speed .asd file"""
+    # temporary try statement until libasd has correct binaries
+    try:
+        load_scan_asd.img_path = str(load_scan_asd.img_paths[0])
+        load_scan_asd.filename = load_scan_asd.img_paths[0].stem
+        images, px_to_nm_scaling = load_scan_asd.load_asd()
+        image = images[40]  # some frames are just black so pick a middle one
+        assert len(images) == 64
+        assert isinstance(image, np.ndarray)
+        assert image.shape == (256, 256)
+        assert image.sum() == 5958870.556640625
+        assert isinstance(px_to_nm_scaling, float)
+        assert px_to_nm_scaling == 1.953125
+    except AssertionError:
```

Comment:
There is an explicit approach to testing for exceptions with PyTest.

```
    with pytest.rauses(AssertionError):
        <action that raises the error>
```

It often requires splitting out assertion tests to their own function (at least that is the approach I take).

I'm not sure the `try: ...  except:` is necessary here though since `libasd` is installed in the GitHub Workflow and both locally on yours and others developments environments.

Reply:
From what I understand from here https://docs.pytest.org/en/7.1.x/how-to/assert.html, it seems as though `pytest.raises()` passes the test if the error is raised. However, this is not what I want.

Here I am not testing that the produced outcome of the function is to raise and exception/error. Instead these tests should be ignored when `libasd` is not installed, and run when it is. So the try's here check and run that.

Although, you have raised a good point that maybe we can include tests for errors e.g. when the channel is wrong to check the error messages.

Reply:
outdated with libasd import now available